web/controllers/api/docs_controller.ex

```diff
@@ -6,21 +6,20 @@ defmodule HexWeb.API.DocsController do
   @uncompressed_max_size 64 * 1024 * 1024
 
   def show(conn, %{"name" => name, "version" => version}) do
-    if (package = HexWeb.Repo.get_by(Package, name: name)) &&
-       (release = HexWeb.Repo.get_by(assoc(package, :releases), version: version)) do
-      if release.has_docs do
-        redirect conn, external: HexWeb.Utils.docs_tarball_url(package, release)
-      else
-        not_found(conn)
-      end
+    package = HexWeb.Repo.get_by(Package, name: name)
```

Comment:
Consider also moving the not_found logic as well. I would likely have a plug that that gets both package and release and halt with not found if they are not there. You can also just call `Repo.get_by!` and let Ecto+Phoenix do its magic (it will render as 404).